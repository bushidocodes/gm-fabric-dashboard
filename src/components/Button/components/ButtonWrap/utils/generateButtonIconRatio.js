/**
 * Helper function that maps button sizes to numbers representing icon ratios
 *
 * @param {string} size
 * @returns number
 */
function mapSizeToButtonIconRatio(size) {
  switch (size) {
    case "xs":
      return 0.5;
    case "sm":
      return 0.75;
    case "lg":
      return 1.25;
    case "xl":
      return 1.5;
    case "normal":
    default:
      return 1;
  }
}

/**
 * Utility function that generates a CSS mixin containing button icon rulesets
 *
 * @param {string} size
 * @returns string
 */
function generateButtonIconRatio(size) {
  const viewboxSize = 24; // default viewbox size 24px
  const ratio = mapSizeToButtonIconRatio(size); // default ratio 1
  const viewboxDimensions = viewboxSize * ratio;
  return `> svg { width: ${/* sc-value */ viewboxDimensions}px; 
  height: ${/* sc-value */ viewboxDimensions}px; }`;
}

export default generateButtonIconRatio;
