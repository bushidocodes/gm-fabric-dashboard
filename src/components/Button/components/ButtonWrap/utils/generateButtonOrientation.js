import { spacingScale } from "style/styleFunctions";

/**
 * Utility function that generates a CSS mixin containing button orientation rulesets
 *
 * @param {string} [alignment="horizontal"]
 * @param {number} [spacingBase=0.5]
 * @returns string
 */
function generateButtonOrientation(
  alignment = "horizontal",
  spacingBase = 0.5
) {
  if (alignment === "horizontal") {
    return `
      flex-direction: row;
      > :not(:first-child) {
        margin-left: ${/* sc-value */ spacingScale(spacingBase)};
        flex-basis: auto;
      }
    `;
  } else if (alignment === "vertical") {
    return `
      flex-direction: column;
      padding-top: ${/* sc-value */ spacingScale(spacingBase)};
      > :not(:first-child) {
        margin-top: ${/* sc-value */ spacingBase / 2}px;
      }
    `;
  }
}

export default generateButtonOrientation;
