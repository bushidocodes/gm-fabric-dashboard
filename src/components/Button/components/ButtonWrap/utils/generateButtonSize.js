import { spacingScale } from "style/styleFunctions";

/**
 * Utility function that transforms an object containing different attributes
 * into a block of CSS attributes dealing with button size
 *
 * @param {any} {
 *   fontSizeMultiplier = 1,
 *   horizontalPaddingMultiplier = 1,
 *   verticalPaddingMultiplier = 1,
 *   textIsUppercase = false
 * }
 * @returns string
 */
function _buttonSizeStyleBlockGenerator({
  fontSizeMultiplier = 1,
  horizontalPaddingMultiplier = 1,
  verticalPaddingMultiplier = 1,
  textIsUppercase = false
}) {
  const fontSizeBase = 14;
  const borderRadiusBase = 3;
  const contentSpacingBase = parseInt(spacingScale(0.5), 10);
  return `
    font-size: ${Math.ceil(fontSizeBase * fontSizeMultiplier)}px;
    border-radius: ${borderRadiusBase}px;
    padding: ${contentSpacingBase *
      horizontalPaddingMultiplier}px ${contentSpacingBase *
    verticalPaddingMultiplier}px;
    ${textIsUppercase ? "text-transform: uppercase;" : ""}
  `;
}

/**
 * Utility function that renders a block of CSS attributes for a
 * particular string (xs, sm, normal, lg, xl)
 *
 * @param {string} size
 * @returns
 */
function generateButtonSize(size) {
  switch (size) {
    case "xs": {
      return _buttonSizeStyleBlockGenerator({
        fontSizeMultiplier: 0.5,
        horizontalPaddingMultiplier: 0.25,
        verticalPaddingMultiplier: 1,
        textIsUppercase: true
      });
    }
    case "sm": {
      return _buttonSizeStyleBlockGenerator({
        fontSizeMultiplier: 0.75,
        horizontalPaddingMultiplier: 0.25,
        verticalPaddingMultiplier: 1,
        textIsUppercase: true
      });
    }
    case "lg": {
      return _buttonSizeStyleBlockGenerator({
        fontSizeMultiplier: 1.25,
        horizontalPaddingMultiplier: 0.5,
        verticalPaddingMultiplier: 1,
        textIsUppercase: false
      });
    }
    case "xl": {
      return _buttonSizeStyleBlockGenerator({
        fontSizeMultiplier: 1.5,
        horizontalPaddingMultiplier: 0.5,
        verticalPaddingMultiplier: 1,
        textIsUppercase: false
      });
    }
    default:
    case "normal": {
      return _buttonSizeStyleBlockGenerator({
        fontSizeMultiplier: 1,
        horizontalPaddingMultiplier: 0.35,
        verticalPaddingMultiplier: 1,
        textIsUppercase: false
      });
    }
  }
}

export default generateButtonSize;
