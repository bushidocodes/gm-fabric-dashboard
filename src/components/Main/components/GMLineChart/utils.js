// importing a single function instead of the whole _
import isEmpty from "lodash/isEmpty";

/**
 * Converts timeseries with time stamps to a string of values without timestamps
 *
 * @param {array} array a timeseries array with time stamps
 * @returns {array}     an array of values
 */
export function numericalTimeSeriesFunc(array) {
  if (array instanceof Array === true && !isEmpty(array)) {
    return array[0].map(cell => cell[1]);
  } else {
    return 0;
  }
}

/**
 * Produces an average of array of values
 *
 * @param {array} array an array of values
 * @returns {number}
 */
export function average(array) {
  if (array instanceof Array === true && !isEmpty(array)) {
    return array.reduce((a, b) => a + b) / array.length;
  } else {
    return 0;
  }
}

/**
 * Produces a median of array of values
 * (if no middle value, average of 2 most central values is returned)
 *
 * @param {array} array an array of values
 * @returns {number}
 */
export function median(array) {
  if (array instanceof Array === true && !isEmpty(array)) {
    const len = array.length;
    let median = 0;
    array.sort();

    if (len % 2 === 0) {
      // if even, average of two middle numbers
      median = (array[len / 2 - 1] + array[len / 2]) / 2;
    } else {
      // if odd, middle number only
      median = array[(len - 1) / 2];
    }
    return median;
  } else {
    return 0;
  }
}

/**
 * Produces a mode(s) of array of values
 * if more than one mode, returns all applicable modes as well as their frequency
 *
 * @param {array} array an array of values
 * @returns {array}     an array that contains mode(s) and frequency value
 */
export function modes(array) {
  if (array instanceof Array === true && !isEmpty(array)) {
    let modes = [];
    let count = [];
    let i;
    let number;
    let maxIndex = 0;

    for (let i = 0; i < array.length; i++) {
      number = array[i];
      count[number] = (count[number] || 0) + 1;
      if (count[number] > maxIndex) {
        maxIndex = count[number];
      }
    }

    for (i in count)
      if (count.hasOwnProperty(i)) {
        if (count[i] === maxIndex) {
          modes.push(Number(i));
        }
      }

    modes.push({ frequency: maxIndex });

    if (maxIndex > 1) {
      return modes;
    } else {
      modes = "no value occurs more than once";
      return modes;
    }
  } else {
    return 0;
  }
}

/**
 * Produces a string that contains descriptive statistics of the graph
 *
 * @param {array} timeSeries an array of values
 * @param {string} title chart title
 * @returns {string}     returns a string with summary chart statistics
 */
export function screenReaderGraphDescription(timeSeries, title, intl) {
  const numericalTimeSeries = numericalTimeSeriesFunc(timeSeries);

  // We don't have negative values in our charts, thus average of 0 means that all values are 0, aka no meaningful data
  if (average(numericalTimeSeries) > 0) {
    let screenReaderGraphData = {
      median: median(numericalTimeSeries).toFixed(2),
      average: average(numericalTimeSeries).toFixed(2),
      mode: modes(numericalTimeSeries),
      max: numericalTimeSeries.sort()[numericalTimeSeries.length - 1],
      min: numericalTimeSeries.sort()[0],
      dataPoints: numericalTimeSeries.length,
      timeSeries: numericalTimeSeries,
      title
    };

    if (
      typeof screenReaderGraphData.mode !== "string" &&
      screenReaderGraphData.mode.length > 1
    ) {
      screenReaderGraphData = Object.assign({}, screenReaderGraphData, {
        mode: `Mode has frequency of ${
          screenReaderGraphData.mode.pop().frequency
        } and values of ${screenReaderGraphData.mode}`
      });
    }

    return intl.formatMessage(
      {
        id: "GMLineChart.screenReaderGraphDescription.withData",
        defaultMessage:
          "A tabular representation of the {title} chart data: median {median} average {average} mode {mode} maximum {max} minimum {min} number of observations {dataPoints} complete data time series follows {timeSeries}",
        description: "Screen reader description for GMLineChart"
      },
      screenReaderGraphData
    );
  } else {
    return intl.formatMessage({
      id: "GMLineChart.screenReaderGraphDescription.noData",
      defaultMessage: `The average for currently displayed data is equal to 0.`,
      description: "Screen reader description for GMLineChart"
    });
  }
}
