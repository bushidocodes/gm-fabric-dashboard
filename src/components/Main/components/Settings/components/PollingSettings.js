import _ from "lodash";
import { PropTypes } from "prop-types";
import React, { Component } from "react";
import InputRange from "react-input-range";

import { contrastColor } from "style/styleFunctions";
import { COLOR_SUCCESS, COLOR_WHITE } from "style/styleVariables";
import Button from "components/Button";
import LayoutSection from "components/LayoutSection";
import PollingBtnContainer from "./components/PollingBtnContainer";
import PollingSliderContainer from "./components/PollingSliderContainer";
import Tooltip from "components/Tooltip";
/**
 * Control to start/stop polling and change the polling rate
 * Styled to resemble a Readout and intended to be a child of SettingsGrid
 */
class PollingSettings extends Component {
  static propTypes = {
    changePollingInterval: PropTypes.func.isRequired,
    glyph: PropTypes.string.isRequired,
    interval: PropTypes.number.isRequired,
    isDisabled: PropTypes.bool,
    isPolling: PropTypes.bool.isRequired,
    startPolling: PropTypes.func.isRequired,
    stopPolling: PropTypes.func.isRequired,
    title: PropTypes.string.isRequired,
    tooltipContent: PropTypes.string
  };

  // Use local state to have a "loosely" controlled component whereby the slider
  // slides smoothly and changes to Redux are debounced.
  state = {
    localInterval: this.props.interval / 1000,
    debouncedSetInterval: _.debounce(this.props.changePollingInterval, 1000)
  };

  render() {
    const {
      isPolling,
      stopPolling,
      startPolling,
      title,
      glyph,
      tooltipContent,
      isDisabled = false
    } = this.props;
    const buttonGlyph = isPolling ? "Pause" : "Play";
    const buttonLabel = isPolling ? "Pause Polling" : "Resume Polling";

    return (
      <LayoutSection title={title} icon={glyph} flex>
        <Tooltip
          content={tooltipContent}
          position="left"
          disabled={!isDisabled}
          containerStyle={{ border: "none" }}
          contentStyle={{ top: "0px" }}
        >
          <PollingBtnContainer isDisabled={isDisabled}>
            <Button
              clickAction={() => {
                if (isPolling) {
                  stopPolling();
                } else {
                  startPolling();
                }
              }}
              glyph={buttonGlyph}
              glyphRatio={3}
              glyphColor={
                isDisabled
                  ? contrastColor(COLOR_WHITE, 0.7).toString()
                  : COLOR_SUCCESS.string()
              }
              disabled={isDisabled}
              label={buttonLabel}
              orientation={"vertical"}
              outline={"raised"}
              size={"xl"}
              tabIndex={0}
              type={"polling"}
              style={{
                width: "100%",
                height: "100%",
                position: "relative",
                zIndex: 0
              }}
              labelStyle={{
                fontSize: "14px",
                position: "absolute",
                bottom: "10px",
                width: "100%",
                left: "0px"
              }}
            />
          </PollingBtnContainer>
        </Tooltip>

        <PollingSliderContainer
          isDisabled={isDisabled}
          id={`ctrl-slider-${title}`}
        >
          <InputRange
            aria-labelledby="polling interval-name"
            disabled={isDisabled}
            maxValue={120}
            minValue={5}
            onChange={value => {
              this.setState({ localInterval: value });
              this.state.debouncedSetInterval(value * 1000);
            }}
            value={this.state.localInterval}
          />
          <span className="label" id={`interval-name-${title}`}>
            {"Polling Interval(s)"}
          </span>
        </PollingSliderContainer>
      </LayoutSection>
    );
  }
}

export default PollingSettings;
