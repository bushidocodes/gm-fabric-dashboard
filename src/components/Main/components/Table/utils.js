import React from "react";
import TableColHeader from "components/Main/components/TableColHeader";
import TableColLatencyHeader from "components/Main/components/TableColLatencyHeader";

/** function getTableHeaders
 * takes Table type and returns appropriate headers with styling
 * for Function/Routes Table or GMServiceTable (Instances Table)
 * @param {object} items
 * @returns {React.element}
 */
export const getTableHeaders = type => {
  let headerNames = [];
  if (type === "Instance") headerNames = [type, "Uptime"];
  if (type === "Route" || type === "Function")
    headerNames = [type, "Requests/sec", "Requests", "Error %", "Latency"];
  let styleProps = {};

  // apply greater flex-basis for first column (20%)
  // requests/sec is a sparkline header and set to align left
  // set text-align to right for additional text columns
  return headerNames.map((header, headerIdx) => {
    styleProps = {};
    if (headerIdx === 0) styleProps = { flex: "1 1 20%" };
    if (headerIdx !== 0 && header !== "Requests/sec")
      styleProps = { textAlign: "right" };

    if (header === "Latency") {
      return <TableColLatencyHeader key={header} />;
    } else {
      return (
        <TableColHeader style={styleProps} key={header}>
          {header}
        </TableColHeader>
      );
    }
  });
};

// simple helper that evaluates and returns route or func props
export const getItem = (route, func) => {
  if (route !== undefined) return route;
  if (func !== undefined) return func;
};
