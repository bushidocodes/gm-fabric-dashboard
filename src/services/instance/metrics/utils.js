import { getState } from "jumpstate";
import { getFabricServer } from "utils/head";

export function buildDiscoveryServiceInstanceMetricsEndpoint(
  fabricServer = getState().fabric.fabricServer || getFabricServer(),
  service = getState().fabric.selectedService || "",
  version = getState().fabric.selectedServiceVersion || "",
  instanceID = getState().fabric.selectedInstance
) {
  return `${fabricServer}/metrics/${service}/${version}/${instanceID}`;
}

/**
 * Clears the interval with the ID stored at window.refreshInstanceMetricsIntervalID and then
 * wipes window.refreshMetricsIntervalID
 */
export function clearInstanceMetricsPollingIntervalIfNeeded() {
  if (window.refreshInstanceMetricsPollingInterval) {
    clearInterval(window.refreshInstanceMetricsPollingInterval);
    window.refreshInstanceMetricsPollingInterval = null;
  }
}
