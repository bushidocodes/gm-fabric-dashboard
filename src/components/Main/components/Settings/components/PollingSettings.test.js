import React from "react";
import { mountWithIntl, renderWithIntl } from "utils/i18nTesting";
import PollingSettings from "./PollingSettings";

let wrapper;
const mockChangePollingInterval = jest.fn();
const mockStartPolling = jest.fn();
const mockStopPolling = jest.fn();
mockStartPolling.mockReturnValue("Start Polling Function");
mockStopPolling.mockReturnValue("Stop Polling Function");

let mockProps = {
  changePollingInterval: mockChangePollingInterval,
  interval: 5000,
  isPolling: true,
  startPolling: mockStartPolling,
  stopPolling: mockStopPolling,
  title: "Instance Polling",
  glyph: "Poll"
};

describe("PollingSettings component", () => {
  beforeEach(function() {
    mockStartPolling.mockClear();
    mockStopPolling.mockClear();
    wrapper = mountWithIntl(<PollingSettings {...mockProps} />);
  });

  test("Matches the snapshot", () => {
    const PollingSettingsComponent = renderWithIntl(
      <PollingSettings {...mockProps} />
    );
    expect(PollingSettingsComponent).toMatchSnapshot();
  });

  test("Resume / Pause button component is rendered", () => {
    expect(wrapper.find("button").length).toBe(1);
  });

  test("If isPolling prop is set to 'true', button displays 'Pause Polling'", () => {
    wrapper.setProps({ isPolling: true });
    expect(wrapper.find("button").props().title).toBe("Pause Polling");
  });

  test("If isPolling prop is set to 'true', and button is clicked, the function that is called is Stop Polling", () => {
    wrapper.setProps({ isPolling: true });
    wrapper.find("button").simulate("click", { preventDefault() {} });
    expect(mockStopPolling).toBeCalled();
  });

  test("If isPolling prop is set to 'false', the PollingSettings Button displays 'Resume Polling'", () => {
    wrapper.setProps({ isPolling: false });
    expect(wrapper.find("button").props().title).toBe("Resume Polling");
  });

  test("If isPolling prop is set to 'false', and button is clicked, the function that is called is Start Polling", () => {
    wrapper.setProps({ isPolling: false });
    wrapper.find("button").simulate("click", { preventDefault() {} });
    expect(mockStartPolling).toBeCalled();
  });

  test("Glyph is rendered as part of the resume/ pause button", () => {
    expect(
      wrapper
        .find("button")
        .find("g")
        .at(1)
        .hasClass("glyph")
    ).toBe(true);
  });

  test("LayoutSection component is rendered by making sure header tag contains 'Instance Polling' text", () => {
    expect(wrapper.find("LayoutSection")).toHaveLength(1);
  });

  test("InputRange component is getting rendered", () => {
    expect(wrapper.find(".input-range").length).toBe(1);
  });

  test("InputRange value depends on the initial state of the Polling Settings component", () => {
    mockProps.interval = 99000;
    const wrapper = mountWithIntl(<PollingSettings {...mockProps} />);
    expect(wrapper.find("InputRange").props().value).toBe(99);
  });
});
