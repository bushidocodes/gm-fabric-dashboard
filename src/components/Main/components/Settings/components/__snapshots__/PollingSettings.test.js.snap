// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PollingSettings component Matches the snapshot 1`] = `
.c4 {
  -webkit-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
}

.c3 {
  z-index: 1;
  height: 24px;
  width: 24px;
  color: currentColor;
  word-spacing: 0;
  vertical-align: middle;
}

.c3:not(:root) {
  overflow: visible;
}

.c8 {
  box-sizing: border-box;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  font-family: Metropolis,-apple-system,BlinkMacSystemFont,"Segoe UI","Roboto","Oxygen","Ubuntu","Cantarell","Fira Sans","Droid Sans","Helvetica Neue",sans-serif;
  font-weight: 600;
  border-width: 1px;
  line-height: 1.4;
  border-style: solid;
  text-transform: none;
  text-align: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  -webkit-transition: all 0.3s ease;
  transition: all 0.3s ease;
  background-color: rgb(255,255,255);
  border-color: rgb(255,255,255);
  border-top-color: hsl(0,0%,100%);
  border-bottom-color: hsl(0,0%,94%);
  color: rgb(0,0,0);
  font-size: 21px;
  border-radius: 3px;
  padding: 2px 4px;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding-top: 4px;
}

.c8:hover {
  -webkit-transition: all 0.1s ease;
  transition: all 0.1s ease;
}

.c8:focus,
.c8:active {
  -webkit-transition: all 0 ease;
  transition: all 0 ease;
}

.c8 > * {
  -webkit-flex: 0 1 auto;
  -ms-flex: 0 1 auto;
  flex: 0 1 auto;
}

.c8:active {
  -webkit-transition-duration: 0s;
  transition-duration: 0s;
  background-color: hsl(0,0%,92.5%);
  border-color: hsl(0,0%,92.5%);
  border-top-color: hsl(0,0%,98%);
  border-bottom-color: hsl(0,0%,87%);
  color: hsl(0,0%,0%);
}

.c8:hover {
  background-color: hsl(0,0%,98.5%);
  border-color: hsl(0,0%,98.5%);
  border-top-color: hsl(0,0%,100%);
  border-bottom-color: hsl(0,0%,92.6%);
  color: hsl(0,0%,0%);
}

.c8:focus {
  outline: none;
  box-shadow: 0 0 0 2px #0aab2a;
  z-index: 1;
}

.c8[disabled],
.c8[disabled]:hover,
.c8[disabled]:focus,
.c8[disabled]:active {
  cursor: default;
  opacity: 0.8;
  background-color: hsl(0,0%,94%);
  border-color: hsl(0,0%,94%);
  color: hsl(0,0%,0%);
}

.c8 > svg {
  width: 24px;
  height: 24px;
}

.c8 > :not(:first-child) {
  margin-top: 0.25px;
}

.c0 {
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  color: rgb(32,32,32);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-bottom: 32px;
}

.c0:first-of-type > * {
  border-top: 0;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-top: 1px solid rgb(217,217,217);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  margin: 0 auto;
  max-width: 1300px;
  padding: 16px;
  width: 100%;
}

.c6 {
  margin: 0 auto;
  max-width: 1300px;
  padding: 8px;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c2 {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  padding-right: 8px !important;
}

.c5 {
  color: inherit;
  font-size: 18px;
  margin: 0;
}

.c7 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-align-self: stretch;
  -ms-flex-item-align: stretch;
  align-self: stretch;
  border: 1px solid rgb(230,230,230);
  border-radius: 3px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 0 0 160px;
  -ms-flex: 0 0 160px;
  flex: 0 0 160px;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 140px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 160px;
}

.c7 .label {
  bottom: 10px;
  position: absolute;
  left: 0;
  width: 100%;
}

.c7 > svg {
  color: rgb(10,171,42);
  -webkit-transition: all 0.3s ease;
  transition: all 0.3s ease;
  will-change: transform;
}

.c7 .btn-type-polling > svg {
  height: 100%;
}

.c7:hover svg,
.c7:active svg,
.c7:focus svg {
  -webkit-transform: scale(1.1);
  -ms-transform: scale(1.1);
  transform: scale(1.1);
  -webkit-transition: all 0.1s ease;
  transition: all 0.1s ease;
}

.c7:active svg {
  -webkit-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
  -webkit-transition: all 0;
  transition: all 0;
}

.c9 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: 1px solid rgb(230,230,230);
  border-radius: 3px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 0 1 50%;
  -ms-flex: 0 1 50%;
  flex: 0 1 50%;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-left: 16px;
  padding: 32px;
  position: relative;
}

.c9 .input-range__label.input-range__label--value {
  color: rgb(10,171,42);
  font-family: Metropolis,-apple-system,BlinkMacSystemFont,"Segoe UI","Roboto","Oxygen","Ubuntu","Cantarell","Fira Sans","Droid Sans","Helvetica Neue",sans-serif;
  font-size: 24px;
  font-weight: 400;
  line-height: 0;
  margin-left: -50%;
  -webkit-transform: translateX(-50%) translateY(4px);
  -ms-transform: translateX(-50%) translateY(4px);
  transform: translateX(-50%) translateY(4px);
  text-align: center;
  will-change: transform;
}

.c9 .input-range__label.input-range__label--value .input-range__label-container {
  position: static;
}

.c9 .label {
  bottom: 10px;
  font-size: 14px;
  left: 0px;
  position: absolute;
  text-align: center;
  width: 100%;
}

.c9 .input-range__slider-container {
  -webkit-transition: left 0;
  transition: left 0;
  will-change: transform;
}

.c9 .input-range__slider {
  background: rgb(10,171,42);
  border: none;
  border-radius: 0;
  -webkit-transition: -webkit-transform 0.3s ease;
  -webkit-transition: transform 0.3s ease;
  transition: transform 0.3s ease;
}

.c9 .input-range__slider:hover,
.c9 .input-range__slider:focus,
.c9 .input-range__slider:active {
  -webkit-transform: scale(1.1);
  -ms-transform: scale(1.1);
  transform: scale(1.1);
}

.c9 .input-range__slider:before {
  border-left: 7px solid transparent;
  border-right: 7px solid transparent;
  border-top: 7px solid rgb(10,171,42);
  color: rgb(10,171,42);
  content: "";
  height: 0;
  position: absolute;
  top: 100%;
  width: 0;
}

.c9 .input-range__track {
  background-color: rgb(230,230,230);
  border-radius: 20px;
}

<section
  className="c0"
>
  <header
    className="c1"
  >
    <span
      className="c2"
    >
      <svg
        aria-labelledby="ariaLabelledby"
        className="c3"
        focusable="false"
        version="1.1"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
        xmlnsXlink="http://www.w3.org/1999/xlink"
      >
        <g
          className="c4"
          fill="currentColor"
          title=""
        >
          <title>
            
          </title>
          <g
            className="glyph"
            fill="currentColor"
            transform="translate(0 0) scale(1)"
          >
            <path
              d="M14.9840176,4.61673196 C17.8864554,5.80181797 19.9317664,8.65279612 19.9317664,11.9814822 C19.9317664,16.3731065 16.3716515,19.9332214 11.9800272,19.9332214 C7.58840298,19.9332214 4.02828809,16.3731065 4.02828809,11.9814822 C4.02828809,8.78101943 5.91906039,6.02217015 8.64450856,4.76103084 L5.27758564,4.46646325 L5.39309226,3.14621652 L10.8941203,3.62749411 L10.4128427,9.12852216 L9.09259598,9.01301553 L9.32170102,6.39433295 C7.22022775,7.39505768 5.79534124,9.54048188 5.79534124,11.9814822 C5.79534124,15.39719 8.56431948,18.1661683 11.9800272,18.1661683 C15.395735,18.1661683 18.1647133,15.39719 18.1647133,11.9814822 C18.1647133,9.69401394 16.9113633,7.64421418 14.9840176,6.57297136 L14.9840176,8.00561267 L13.7470804,8.00561267 L13.7470804,3.14621652 L14.9840176,3.14621652 L14.9840176,4.61673196 L14.9840176,4.61673196 Z"
              id="Combined-Shape"
            />
          </g>
        </g>
      </svg>
    </span>
    <h3
      className="c5"
    >
      Instance Polling
    </h3>
  </header>
  <div
    className="c6"
  >
    <div
      className="c7"
    >
      <button
        className="c8"
        disabled={undefined}
        onClick={[Function]}
        orientation="vertical"
        size="xl"
        style={
          Object {
            "height": "100%",
            "position": "relative",
            "width": "100%",
            "zIndex": 0,
          }
        }
        tabIndex={0}
        title="Pause Polling"
        type="polling"
      >
        <svg
          aria-labelledby="ariaLabelledby"
          className="c3"
          focusable="false"
          version="1.1"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
          xmlnsXlink="http://www.w3.org/1999/xlink"
        >
          <g
            className="c4"
            fill="currentColor"
            title=""
          >
            <title>
              
            </title>
            <g
              className="glyph"
              fill="rgb(10, 171, 42)"
              transform="translate(-24 -24) scale(3)"
            >
              <path
                d="M6,5 L10.6153846,5 L10.6153846,19 L6,19 L6,5 Z M13.3846154,5 L18,5 L18,19 L13.3846154,19 L13.3846154,5 Z"
                id="Combined-Shape"
              />
            </g>
          </g>
        </svg>
        <span
          style={
            Object {
              "bottom": "10px",
              "fontSize": "14px",
              "left": "0px",
              "position": "absolute",
              "width": "100%",
            }
          }
        >
          
          Pause Polling
          
        </span>
      </button>
    </div>
    <div
      className="c9"
      id="ctrl-slider-Instance Polling"
    >
      <div
        aria-disabled={false}
        className="input-range"
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onTouchStart={[Function]}
      >
        <span
          className="input-range__label input-range__label--min"
        >
          <span
            className="input-range__label-container"
          >
            5
          </span>
        </span>
        <div
          className="input-range__track input-range__track--background"
          onMouseDown={[Function]}
          onTouchStart={[Function]}
        >
          <div
            className="input-range__track input-range__track--active"
            style={
              Object {
                "left": "0%",
                "width": "0%",
              }
            }
          />
          <span
            className="input-range__slider-container"
            style={
              Object {
                "left": "0%",
                "position": "absolute",
              }
            }
          >
            <span
              className="input-range__label input-range__label--value"
            >
              <span
                className="input-range__label-container"
              >
                5
              </span>
            </span>
            <div
              aria-controls={undefined}
              aria-labelledby={undefined}
              aria-valuemax={120}
              aria-valuemin={5}
              aria-valuenow={5}
              className="input-range__slider"
              draggable="false"
              onKeyDown={[Function]}
              onMouseDown={[Function]}
              onTouchStart={[Function]}
              role="slider"
              tabIndex="0"
            />
          </span>
        </div>
        <span
          className="input-range__label input-range__label--max"
        >
          <span
            className="input-range__label-container"
          >
            120
          </span>
        </span>
      </div>
      <span
        className="label"
        id="interval-name-Instance Polling"
      >
        Polling Interval (s)
      </span>
    </div>
  </div>
</section>
`;
